packages/manager/apps/telecom/src/app/telecom/pack/access-list/xdsl-access-list.controller.js

```diff
@@ -0,0 +1,118 @@
+import {
+  ELIGIBILITY,
+  PAGE_SIZE,
+  ACCESS_TYPE,
+  URL_CLOSURE_INFO,
+} from './xdsl-access-list.constant';
+
+export default class XdslAccessListCtrl {
+  /* @ngInject */
+  constructor(coreURLBuilder, XdslAccessListService, TucToastError) {
+    this.coreURLBuilder = coreURLBuilder;
+    this.XdslAccessListService = XdslAccessListService;
+    this.TucToastError = TucToastError;
+  }
+
+  $onInit() {
+    this.ELIGIBILITY = ELIGIBILITY;
+    this.pageSize = PAGE_SIZE;
+    this.ACCESS_TYPE = ACCESS_TYPE;
+
+    this.links = {
+      copperClosureMore: URL_CLOSURE_INFO,
+    };
+
+    this.getServices();
+  }
+
+  getServices() {
+    this.services = [];
+    this.XdslAccessListService.getServices()
+      .then(({ data }) => this.buildServicesList(data || []))
+      .catch(({ data }) => {
+        this.TucToastError(data.message);
+      });
+  }
+
+  buildServicesList(services) {
+    // Retrieve service info for each service
+    services.forEach((serviceName) => {
+      this.XdslAccessListService.getServiceDetail(serviceName).then(
```

Comment:
If i understand we have 1 call per service. It isn't the good practice.
You have to use a iceberg call for to have directly the service with their detail. We haven't to 1 call for id and x call for the detail